src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1584 @@
+              together all input models' error arrays.
+
+            Error array will be assigned to ``'err'`` key of the output model.
+
+            .. note::
+                At this time, output error array is not equivalent to
+                error propagation results.
+
+        """
+        # always required:
+        attributes = {
+            "data",
+            "wcs",
+            "wht",
+        }
+
+        if enable_ctx:
+            attributes.add("con")
+        if compute_err:
+            attributes.add("err")
+        if enable_var:
+            attributes.update(
+                ["var_rnoise", "var_poisson", "var_flat"]
+            )
+            # TODO: if we want to support adding more data to
+            # existing output models, we need to also store weights
+            # for variance arrays:
+            # var_rnoise_weight
+            # var_flat_weight
+            # var_poisson_weight
+        if accumulate:
+            if enable_ctx:
+                attributes.add("n_coadds")
+
+            # additional attributes required for input parameter 'output_model'
+            # when data and weight arrays are not None:
+            attributes.update(
+                {
+                    "pixfrac",
+                    "kernel",
+                    "fillval",
+                    "weight_type",
+                    "pointings",
+                    "exposure_time",
+                    "measurement_time",
+                    "start_time",
+                    "end_time",
+                    "duration",
+                }
+            )
+
+        return attributes
+
+    def check_memory_requirements(self, output_model, allowed_memory,
+                                  n_input_models=None):
+        """ Called just before `create_output_model` returns to verify
+        that there is enough memory to hold the output.
+
+        Parameters
+        ----------
+        allowed_memory : float, None
+            Fraction of memory allowed to be used for resampling. If
+
+        output_model : dict, None, optional
+            A dictionary containing data arrays and other attributes that
+            will be used to add new models to. use
+            :py:meth:`Resample.output_model_attributes` to get the list of
+            keywords that must be present. When ``accumulate`` is `False`,
+            only the WCS object of the model will be used. When ``accumulate``
+            is `True`, new models will be added to the existing data in the
+            ``output_model``.
+
+            When ``output_model`` is `None`, a new model will be created.
+
+        n_input_models : int, None, optional
+            Number of input models expected to be resampled. When provided,
+            this is used to estimate memory requirements and optimize memory
+            allocation for the context array.
+
+
+        """
+        if ((allowed_memory is None and
+                "DMODEL_ALLOWED_MEMORY" not in os.environ) or
```

Comment:
This is no longer used in jwst or in roman. See:
https://github.com/spacetelescope/jwst/pull/8975
I suggest that we don't re-introduce this and I think removing it (and the allowed memory check) would allow dropping `psutil` as a runtime dependency.